packages/main/src/SplitButton.js

```diff
@@ -0,0 +1,449 @@
+	isTabNext,
+	isTabPrevious,
+	isEscape,
+	isSpace,
+	isEnter,
+	isDown,
+	isUp,
+	isDownAlt,
+	isUpAlt,
+	isF4,
+	isShift,
+} from "@ui5/webcomponents-base/dist/Keys.js";
+import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import SplitButtonTemplate from "./generated/templates/SplitButtonTemplate.lit.js";
+import ButtonDesign from "./types/ButtonDesign.js";
+
+import {
+	SPLIT_BUTTON_DESCRIPTION,
+	SPLIT_BUTTON_KEYBOARD_HINT,
+} from "./generated/i18n/i18n-defaults.js";
+
+// Styles
+import SplitButtonCss from "./generated/themes/SplitButton.css.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-split-button",
+	properties: /** @lends sap.ui.webcomponents.main.SplitButton.prototype */ {
+		/**
+		 * Defines the text of the component.
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 */
+		text: {
+			type: String,
+		},
+
+		/**
+		 * Defines the icon to be displayed as graphical element within the component.
+		 * The SAP-icons font provides numerous options.
+		 * <br><br>
+		 * Example:
+		 *
+		 * See all the available icons in the <ui5-link target="_blank" href="https://openui5.hana.ondemand.com/test-resources/sap/m/demokit/iconExplorer/webapp/index.html" class="api-table-content-cell-link">Icon Explorer</ui5-link>.
+		 *
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 */
+		icon: {
+			type: String,
+		},
+
+		/**
+		 * Defines the icon to be displayed in active state as graphical element within the component.
+		 *
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 */
+		activeIcon: {
+			type: String,
+		},
+
+		/**
+		 * Defines the component design.
+		 *
+		 * <br><br>
+		 * <b>The available values are:</b>
+		 *
+		 * <ul>
+		 * <li><code>Default</code></li>
+		 * <li><code>Emphasized</code></li>
+		 * <li><code>Positive</code></li>
+		 * <li><code>Negative</code></li>
+		 * <li><code>Transparent</code></li>
+		 * <li><code>Attention</code></li>
+		 * </ul>
+		 *
+		 * @type {ButtonDesign}
+		 * @defaultvalue "Default"
+		 * @public
+		 */
+		design: {
+			type: ButtonDesign,
+			defaultValue: ButtonDesign.Default,
+		},
+
+		/**
+		 * Defines whether the component is disabled.
+		 * A disabled component can't be pressed or
+		 * focused, and it is not in the tab chain.
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		disabled: {
+			type: Boolean,
+		},
+
+		/**
+		 * Indicates if the elements is on focus
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @private
+		 */
+		focused: {
+			type: Boolean,
+		},
+
+		/**
+		 * Accessibility-related properties for inner elements of the Split Button
+		 *
+		 * @type {Object}
+		 * @private
+		 */
+		_splitButtonAccInfo: {
+			type: Object,
+		},
+
+		/**
+		 * Sets the accessible aria name of the component.
+		 *
+		 * @type {String}
+		 * @defaultvalue: ""
+		 * @public
+		 */
+		 accessibleName: {
```

Comment:
minor: this can be brought up to have better order first the public props and then the privates ones, not a bit of a disorder as now.

Reply:
Yeah, sure... moved.